pennylane/transforms/compile.py

```diff
@@ -0,0 +1,148 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Code for the high-level quantum function transform that executes compilation."""
+
+
+from pennylane import apply
+from pennylane.tape import get_active_tape
+from pennylane.ops import __all__ as all_ops
+
+from pennylane.transforms import qfunc_transform
+from pennylane.transforms.optimization import cancel_inverses, commute_controlled, merge_rotations
+
+
+default_pipeline = [commute_controlled, cancel_inverses, merge_rotations]
```

Comment:
I'm curious why this order of transformations is chosen?

Reply:
Seemed like a reasonable choice in terms of minimizing the size of the tape to loop through - first, push everything through as much as possible, then remove the inverses, and only merge what's left. Alternatively, you could remove the inverses first - but pushing commuting gates through controls is likely to result in more adjacent inverses being produced.